### ISSUE TITLE
Forecasting: cached RevIN statistics leak across concurrent requests

### ISSUE THREAD

**Issue by PansulBhatt** [NVIDIA/NV-Tesseract#69]
## Summary

Concurrent forecasting requests can use another request's RevIN mean and
standard deviation when they share a cached model instance. The SDK caches
models globally, while `RevIN` stores per-input statistics on the module as
mutable `self.mean` and `self.stdev` fields.

## Impact

The affected request can be returned on the wrong scale without raising an
error. This is especially risky for a service that calls `perform_forecasting`
from multiple threads: the output can look numerically plausible while
belonging to a different request's normalization context.

## Reproduction

The race can be made deterministic by pausing request A after normalization,
letting request B normalize on the same `RevIN` instance, and then allowing
request A to denormalize:

```python
from concurrent.futures import ThreadPoolExecutor
import threading
import torch

from backbone import RevIN

revin = RevIN(num_features=1)
a_normalized = threading.Event()
b_normalized = threading.Event()

def round_trip(x):
    if x.mean().item() >= 50:
        assert a_normalized.wait(timeout=5)

    normalized = revin(x, mode="norm")

    if x.mean().item() < 50:
        a_normalized.set()
        assert b_normalized.wait(timeout=5)
    else:
        b_normalized.set()

    return revin(normalized, mode="denorm")

a = torch.tensor([[[1.0, 2.0, 3.0, 4.0]]])
b = torch.tensor([[[100.0, 200.0, 300.0, 400.0]]])

with ThreadPoolExecutor(max_workers=2) as pool:
    out_a, out_b = list(pool.map(round_trip, (a, b)))
```

Expected: `out_a` equals `a` and `out_b` equals `b`.

Actual: `out_a` is approximately `[[[100, 200, 300, 400]]]`, because request A
is denormalized using request B's statistics.

## Root cause

`RevIN.forward(..., mode="norm")` calls `_get_statistics`, which writes
`self.mean` and `self.stdev`. A later `mode="denorm"` call reads those shared
fields. The forecasting path has encoder and head work between those two
operations, giving another request time to overwrite them.

`sdk.forecasting._load_cached_model` returns the same model object for matching
checkpoint/configuration keys, so concurrent SDK calls can reach the same
`RevIN` instance.

## Proposed fix

Make RevIN statistics invocation-local:

- return `(mean, stdev)` from the normalization step;
- pass that exact pair to the matching denormalization step;
- remove mutable per-request statistics from the module;
- update the integrated-gradients statistics hook to return statistics rather
  than mutating the module;
- add a deterministic concurrent regression test plus sequential and gradient
  behavior coverage.

This preserves model weights and numerical behavior for sequential inference
while making concurrent requests independent.
